pytest_mpl/plugin.py

```diff
@@ -44,26 +43,59 @@
 
 import pytest
 
-from pytest_mpl.summary.html import generate_summary_basic_html, generate_summary_html
+from .kernels import (DEFAULT_HAMMING_TOLERANCE, DEFAULT_HASH_SIZE,
+                      DEFAULT_HIGH_FREQUENCY_FACTOR, KERNEL_SHA256, kernel_factory)
+from .summary.html import generate_summary_basic_html, generate_summary_html
 
+#: The default matplotlib backend.
+DEFAULT_BACKEND = "agg"
+
+#: The default algorithm to use for image hashing.
+DEFAULT_KERNEL = KERNEL_SHA256
+
+#: The default pytest-mpl marker.
+DEFAULT_MARKER = "mpl_image_compare"
+
+#: The default option to remove text from matplotlib plot.
+DEFAULT_REMOVE_TEXT = False
+
+#: The default image RMS tolerance.
+DEFAULT_RMS_TOLERANCE = 2
+
+#: The default matplotlib plot style.
+DEFAULT_STYLE = "classic"
+
+#: Metadata entry in the JSON hash library defining the source kernel of the hashes.
+META_HASH_LIBRARY_KERNEL = 'pytest-mpl-kernel'
+
+#: Valid formats for generate summary.
 SUPPORTED_FORMATS = {'html', 'json', 'basic-html'}
 
-SHAPE_MISMATCH_ERROR = """Error: Image dimensions did not match.
-  Expected shape: {expected_shape}
-    {expected_path}
-  Actual shape: {actual_shape}
-    {actual_path}"""
+#: Supported INET protocols.
+PROTOCOLS = ('http://', 'https://')
 
+#: Template error message for image shape conformance.
+TEMPLATE_SHAPE_MISMATCH = """Error! Image dimensions did not match.
 
-def _hash_file(in_stream):
-    """
-    Hashes an already opened file.
-    """
-    in_stream.seek(0)
-    buf = in_stream.read()
-    hasher = hashlib.sha256()
-    hasher.update(buf)
-    return hasher.hexdigest()
+  Baseline Shape:
+    {baseline_shape}
+  Baseline Image:
+    {baseline_image}
+  Result Shape:
+    {result_shape}
+  Result Image:
+    {result_image}"""
+
+TEMPLATE_IMAGE_DIFFERENCE = """Failed! Image files did not match.
+
+  RMS: {rms}
+  Tolerance: {tol}
+  Baseline Image:
+    {expected}
+  Result Image:
+    {actual}
+  Difference Image:
+    {diff}"""
```

Comment:
Applied the `SHAPE_MISMATCH` template approach here also, which for me, really makes the code readable where it's used.